csharp/src/lesson01/README.md

```diff
@@ -0,0 +1,301 @@
+{
+    internal class Hello
+    {
+        private readonly ITracer _tracer;
+
+        public Hello(OpenTracing.ITracer tracer)
+        {
+            _tracer = tracer;
+        }
+
+        public void SayHello(string helloTo)
+        {
+            var span = _tracer.BuildSpan("say-hello").Start();
+            var helloString = $"Hello, {helloTo}!";
+            Console.WriteLine(helloString);
+            span.Finish();
+        }
+
+        public static void Main(string[] args)
+        {
+            if (args.Length != 1)
+            {
+                throw new ArgumentException("Expecting one argument");
+            }
+
+            var helloTo = args[0];
+            new Hello(GlobalTracer.Instance).SayHello(helloTo);
+        }
+    }
+}
+```
+
+We are using the following basic features of the OpenTracing API:
+  * a `ITracer` instance is used to create a span builder via `BuildSpan()`
+  * each `ISpan` is given an _operation name_, `"say-hello"` in this case
+  * builder is used to create a span via `Start()`
+  * each `ISpan` must be finished by calling its `Finish()` function
+  * the start and end timestamps of the span will be captured automatically by the tracer implementation
+
+However, if we run this program, we will see no difference, and no traces in the tracing UI.
+That's because the `GlobalTracer.Instance` returns a no-op tracer by default.
+
+### Use `Microsoft.Extensions.Logging` for output
+
+In the first draft, we used `Console.WriteLine` for the output. This works very well if there is no multi-threading. 
+As soon as it gets more complex, our output could become unreadable if different sources write to our console at the same time.
+
+Microsoft offers a framework for logging which has it's main target in the ASP.NET world, but also works fine with normal applications. 
+You can find it as NuGet package `Microsoft.Extensions.Logging` which is already added to our project.
+
+We need to create an `ILoggerFactory` instance in our `Main` method. `new LoggerFactory().AddConsole()` will create one that uses 
+the `Console` as logging destination.
+
+An `ILogger` instance can then be used for logging to that destination. `loggerFactory.CreateLogger<Hello>()` creates one that 
+logs to our logging destination, using the `Hello` class as identifier. We store that in `_logger` for later usage.
+
+`Console.WriteLine` can then be replaced by `_logger.LogInformation` which creates an information log.
+
+```csharp
+using System;
+
+namespace OpenTracing.Tutorial.Lesson01.Exercise
+{
+    internal class Hello
+    {
+		...
+        private readonly ILogger<Hello> _logger;
+
+        public Hello(ITracer tracer, ILoggerFactory loggerFactory)
+        {
+            ...
+            _logger = loggerFactory.CreateLogger<Hello>();
+        }
+
+        public void SayHello(string helloTo)
+        {
+            ...
+            _logger.LogInformation(helloString);
+            span.Log("WriteLine");
```

Comment:
should this be here?

Reply:
Yes, but we can also change the log text if you don’t like it